### PR TITLE
Add HDF5 input and output files to eigensolver miniapp

### PR DESCRIPTION
**Pull request by RMeli**
Add HDF5 input and output files to eigensolver miniapp.

---

* [ ] Decide how to call the input matrix dataset
* [ ] Decide how to call the eigenvalues vector dataset
* [ ] Decide how to call the eigenvectors matrix dataset

I used `/a` temporarily (it's what I had in an HDF5 I'm using), but I'm open to better suggestions. Maybe `/input` would do.

### CODE REVIEW

File: miniapp/miniapp_eigensolver.cpp [hunk 1]
```diff
@@ -176,6 +215,10 @@ int main(int argc, char** argv) {
   desc_commandline.add_options()
     ("matrix-size",  value<SizeType>()   ->default_value(4096), "Matrix size")
     ("block-size",   value<SizeType>()   ->default_value( 256), "Block cyclic distribution size")
+#ifdef DLAF_WITH_HDF5
+    ("input-file",   value<std::string>()                     , "Load matrix from given HDF5 file")
+    ("output-file",  value<std::string>()                     , "Save eigenvector and eigenvalue to given HDF5 file")
```
Comment (RMeli) [inline miniapp/miniapp_eigensolver.cpp:220]:
```suggestion
    ("output-file",  value<std::string>()                     , "Save eigenvectors and eigenvalues to given HDF5 file")
```

**Review by msimberg — APPROVED** [review]
Looks reasonable to me.

File: miniapp/miniapp_eigensolver.cpp [hunk 2]
```diff
@@ -176,6 +223,10 @@ int main(int argc, char** argv) {
   desc_commandline.add_options()
     ("matrix-size",  value<SizeType>()   ->default_value(4096), "Matrix size")
     ("block-size",   value<SizeType>()   ->default_value( 256), "Block cyclic distribution size")
+#ifdef DLAF_WITH_HDF5
+    ("input-file",   value<std::string>()                     , "Load matrix from given HDF5 file")
+    ("output-file",  value<std::string>()                     , "Save eigenvectors and eigenvalues to given HDF5 file")
```
Comment (albestro) [inline miniapp/miniapp_eigensolver.cpp:228]: I don't remember why I didn't use `std::filesystem::path` in the beginning (I mean in `miniapp_tridiag_solver`), probably just forgot about it. 😅

File: miniapp/miniapp_tridiag_solver.cpp [hunk 3]
```diff
@@ -9,6 +9,7 @@
 //
 
 #include <cstdlib>
+#include <filesystem>
 #include <iostream>
 #include <limits>
 #include <optional>
```
Comment (RMeli) [inline miniapp/miniapp_tridiag_solver.cpp:15]:
```suggestion
```

**Review by rasolca — APPROVED** [review]

**Comment by rasolca** [comment]
cscs-ci run

**Comment by RMeli** [comment]
cscs-ci run

**Comment by RMeli** [comment]
cscs-ci run

**Review by msimberg — APPROVED** [review]

**Comment by RMeli** [comment]
> cscs-ci run

HDF5 doesn't seem to be enabled in CI, is it? I added `+hdf5` to the CI specs in #980, where I plan to use it for tests (TBD), but maybe it should be added here already?

**Comment by RMeli** [comment]
cscs-ci run

**Review by rasolca — APPROVED** [review]

**Pull request merged**